Move implemented (copy / paste interface in the demo).

OndrejSpanel opened the pull request:
Implemented a move in the tree model. While move can be replaced with delete / insert, this has some serious drawbacks:
- it is hard to validate the move (and to prevent it when the move should not be done)
- it is hard to handle failures (second operation can fail, while the first has already completed)

This pull request adds move into the TreeModel interface, it implements move in the internal tree model and provides a demo implementation for both internal and external model demos (using Cut / Paste interface).

benhutchison commented:
@OndrejSpanel sorry I just noticed this one again while I was scanning for your recent 2.13 PR. Let me know if you still want this merged...

OndrejSpanel commented:
For me it makes little difference, as I have forked the tree component anyway and I am using it from a different location. Other than than, I think the changes make the component much easier to use, therefore merging them is probably a good think.

I have no idea how many people can be still using the components - is there a way to know - some Maven download stats, or anything like that?

OndrejSpanel commented:
I have added some more commits adding drag-drop functionality on top of the move introduced in this PR before. There are also fixes for some issues.

I had these commits sitting in the local Git repository, I have rebased them in case you want to merge them.

BenHutchisonSeek commented:
For myself, Im not using Swing anymore, this is maintained just for legacy support of existing users, and I think they are few. 

Since AFAICT there's some fixes in this PR, Im going to merge it
